lib/server/routes/frames.js

```diff
@@ -318,14 +318,15 @@ FramesRouter.prototype.addShardToFrame = function(req, res, next) {
         }
 
         let mirror = null;
-
         if (mirrors && mirrors.length) {
           const filtered = mirrors.filter((m) => {
-            if (!m.contact) {
-              log.warn('Mirror %s is missing contact in database', m._id);
-              return false;
-            } else if (!m.isEstablished) {
-              return true;
+            if (!bl.includes(m)) {
+              if (!m.contact) {
```

Comment:
May need to be m.contact.nodeID. Because contact may or may not be there, which this section is removing, it will likely be better to do this later with a filter function.